controllers/clustergroupupgrade_controller.go

```diff
@@ -791,20 +821,35 @@ func (r *ClusterGroupUpgradeReconciler) doManagedPoliciesExist(
 			if err != nil {
 				// If the parent policy was not found, add its name to the list of missing policies.
 				if errors.IsNotFound(err) {
-					managedPoliciesMissing = append(managedPoliciesMissing, managedPolicyName)
+					managedPoliciesInfo.missingPolicies = append(managedPoliciesInfo.missingPolicies, managedPolicyName)
 					continue
 				} else {
 					// If another error happened, return it.
-					return false, nil, nil, err
+					return false, managedPoliciesInfo, err
 				}
 			}
 
+			// If the parent policy has invalid hub template, add its name to the list of invalid policies.
+			if policyInvalidHubTmpl[managedPolicyName] {
+				r.Log.Error(&utils.PolicyErr{ObjName: managedPolicyName, ErrMsg: utils.PlcHasHubTmplErr}, "Policy is invalid")
+				managedPoliciesInfo.invalidPolicies = append(managedPoliciesInfo.invalidPolicies, managedPolicyName)
+				continue
+			}
+
+			// If the parent policy is not valid due to missing field, add its name to the list of invalid policies.
+			policyErr := utils.VerifyPolicyObjects(foundPolicy)
+			if policyErr != nil || policyInvalidHubTmpl[managedPolicyName] {
```

Comment:
The test for policyInvalidHubTmpl here is redundant with the test on line 833. Is there a reason it was included here as well?

Reply:
Good catch! That was added by mistake.  Will update

Reply:
Done